### DISCUSSION TITLE
初回リリースに必要な機能

### DISCUSSION THREAD

**Discussion by aca-ta** [Code-for-Funabashi/map#13] [General]
初回リリースに必要な機能を考える

とりあえず思いついたものをばしばし書いて、後で削る感じで考えましょう！

**Comment by aca-ta** [comment]
バシバシと書きましたが、最初はMVP(Minimum Viable Product)を作って、ユーザー(直近は家族や友人等を想定)にヒアリングしながら要求を拾い、細かく機能拡充したいなと思ってます。
なので上げていただいた案のうち最小のものだけを初回機能として実装し、残りは今後の開発アイテムとして今後のマイルストーンに組み込むイメージです！

**Comment by j-takurou** [comment]
>最初はMVP(Minimum Viable Product)を作って、ユーザー(直近は家族や友人等を想定)にヒアリングしながら要求を拾い、細かく機能拡充したいなと思ってます。

認識理解致しました。ありがとうございます!!!
（アジャイル開発経験が乏しいので、いろいろ勉強させていただいております。）

いくつか挙げていきます。初回機能として実装すべき範囲に関して認識合わせさせていただければと思います。

### 【地図上に表示するもの】
1. 指定した地点を中心にX km以内の領域(半透明ポリゴンデータなどで)を表示
2. 指定した地点を中心にX km以内に存在する施設(*)を地図上に表示する
3. 施設タイプごとに異なるMarkerで表示できるようにする

### 【地図以外の要素】
4. HPとしての体裁を整えたいかなと思います。(越谷の子育てマップを念頭においております。[link](http://koshigaya-kosodate.net/maps/?area=5&ctg=))
    - タイトル
    - ヘッダー
    - 地図上に表示した施設の情報(*)をテーブルでも表示する

**Reply by aca-ta** [reply]
あげていただいた

> 指定した地点を中心にX km以内の領域(半透明ポリゴンデータなどで)を表示
> 指定した地点を中心にX km以内に存在する施設(*)を地図上に表示する

これはめちゃめちゃやりたいですが、それなりに実装工数がかかりそうですし、将来的な機能で良いかなと思いましたが、いかがでしょう？

**Reply by j-takurou** [reply]
そうですね...MVPスコープの基準も[こちら](https://github.com/Code-for-Funabashi/kosodate-map/discussions/13#discussioncomment-486544)で明確になった気がします。
”あとでやる”でいい気がしてます！

**Comment by aca-ta** [comment]
現状GoogleMapや市役所の地図で実現できない機能かつ、最低限価値がありそうな機能 + ユーザー導入に最低限必要なものをイメージして書いてみました。

## 地図上に表示する物
- 2つ以上の種類の施設またはポリゴン
    - 色やアイコンで施設の種類がわかる
    - 施設の名称がわかる(tooltip等)

## 地図以外の要素
- kosodate-mapの目的や利用イメージがわかる簡単なドキュメント(100文字程度)
- データソースがわかるドキュメント
    - ~は船橋市オープンデータから、~は国交省から取得してますがわかるもの

**Reply by j-takurou** [reply]
なるほど、確かに必要最低限だとこれか、、（他は思いつかないですね。）

>地図以外の要素

データのreferenceを明記する点抜けてました！ありがとうございます。

**Comment by aca-ta** [comment]
特に論点なければ、以下で確定しようと思いますがいかがでしょう？

## 地図上に表示する物
- 2つ以上の種類の施設またはポリゴンを表示する
- 色やアイコンで施設の種類がわかるようにする
- 施設の名称がわかるようにする(tooltip等)

## 地図以外の要素
- HPとしての体裁を整える
    - タイトル
    - ヘッダー
- kosodate-mapの目的や利用イメージがわかる簡単なドキュメント(100文字程度)
- データソースがわかるドキュメント
    - ~は船橋市オープンデータから、~は国交省から取得してますがわかるもの

**Reply by aca-ta** [reply]
こちらで確定する

**Reply by aca-ta** [reply]
プロジェクト化した
https://github.com/Code-for-Funabashi/kosodate-map/projects/1
